extern/decNumber/decCommon.c

```diff
@@ -254,6 +255,8 @@ static decFloat * decFinalize(decFloat *df, bcdnum *num,
   // decShowNum(num, "final");
   #endif
 
+  assert(umsd<=ulsd);
```

Comment:
I see you've decided to use assert()?

Reply:
Yes. I thought - a porridge can't be spoilt with butter.

Reply:
Do you run tests with the debug build of server?

Reply:
Standard assert() used to write message to stdout so for Firebird it worse than useless: stdout is redirected to network stream in Classic mode.

Reply:
Standard `assert`, as a standard thing, is used in every project that do not want to rewrite the standard, like all libraries we use: re2, libcds, decNumber, etc. So why not? I'd say the changes to decNumber must use (this very ugly) decNumber's code style and must be redirected to some of its project who claims to be the "official" one.

Reply:
> Do you run tests with the debug build of server?

Not on the regular basis.